cli/initiator/reshare.go

```diff
@@ -0,0 +1,231 @@
+		▒██▀ ██▌ ██▄█▒  ██▒ ▀█▒   ▓██ ▒ ██▒▓█   ▀ ▒██    ▒ ▓██░ ██▒▒████▄    ▓██ ▒ ██▒▓█   ▀ 
+		░██   █▌▓███▄░ ▒██░▄▄▄░   ▓██ ░▄█ ▒▒███   ░ ▓██▄   ▒██▀▀██░▒██  ▀█▄  ▓██ ░▄█ ▒▒███   
+		░▓█▄   ▌▓██ █▄ ░▓█  ██▓   ▒██▀▀█▄  ▒▓█  ▄   ▒   ██▒░▓█ ░██ ░██▄▄▄▄██ ▒██▀▀█▄  ▒▓█  ▄ 
+		░▒████▓ ▒██▒ █▄░▒▓███▀▒   ░██▓ ▒██▒░▒████▒▒██████▒▒░▓█▒░██▓ ▓█   ▓██▒░██▓ ▒██▒░▒████▒
+		▒▒▓  ▒ ▒ ▒▒ ▓▒ ░▒   ▒    ░ ▒▓ ░▒▓░░░ ▒░ ░▒ ▒▓▒ ▒ ░ ▒ ░░▒░▒ ▒▒   ▓▒█░░ ▒▓ ░▒▓░░░ ▒░ ░
+		░ ▒  ▒ ░ ░▒ ▒░  ░   ░      ░▒ ░ ▒░ ░ ░  ░░ ░▒  ░ ░ ▒ ░▒░ ░  ▒   ▒▒ ░  ░▒ ░ ▒░ ░ ░  ░
+		░ ░  ░ ░ ░░ ░ ░ ░   ░      ░░   ░    ░   ░  ░  ░   ░  ░░ ░  ░   ▒     ░░   ░    ░   
+		░    ░  ░         ░       ░        ░  ░      ░   ░  ░  ░      ░  ░   ░        ░  ░
+		░`)
+		err := cli_utils.SetViperConfig(cmd)
+		if err != nil {
+			return err
+		}
+		logger, err := cli_utils.SetGlobalLogger(cmd, "dkg-initiator")
+		if err != nil {
+			return err
+		}
+		// workaround for https://github.com/spf13/viper/issues/233
+		if err := viper.BindPFlag("outputPath", cmd.Flags().Lookup("outputPath")); err != nil {
+			logger.Fatal("😥 Failed to bind a flag: ", zap.Error(err))
+		}
+		outputPath := viper.GetString("outputPath")
+		if outputPath == "" {
+			logger.Fatal("😥 Failed to get deposit result path flag value: ", zap.Error(err))
+		}
+		if stat, err := os.Stat(outputPath); err != nil || !stat.IsDir() {
+			logger.Fatal("😥 Error to to open path to store results", zap.Error(err))
+		}
+		// Load operators TODO: add more sources.
+		if err := viper.BindPFlag("operatorsInfo", cmd.Flags().Lookup("operatorsInfo")); err != nil {
+			logger.Fatal("😥 Failed to bind a flag: ", zap.Error(err))
+		}
+		operatorsInfo := viper.GetString("operatorsInfo")
+		if err := viper.BindPFlag("operatorsInfoPath", cmd.Flags().Lookup("operatorsInfoPath")); err != nil {
+			logger.Fatal("😥 Failed to bind a flag: ", zap.Error(err))
+		}
+		operatorsInfoPath := viper.GetString("operatorsInfoPath")
+		if operatorsInfo == "" && operatorsInfoPath == "" {
+			logger.Fatal("😥 Operators string or path have not provided")
+		}
+		if operatorsInfo != "" && operatorsInfoPath != "" {
+			logger.Fatal("😥 Please provide either operator info string or path, not both")
+		}
+		var opMap initiator.Operators
+		if operatorsInfo != "" {
+			logger.Info("📖 reading raw JSON string of operators info")
+			opMap, err = initiator.LoadOperatorsJson([]byte(operatorsInfo))
+			if err != nil {
+				logger.Fatal("😥 Failed to load operators: ", zap.Error(err))
+			}
+		}
+		if operatorsInfoPath != "" {
+			opMap, err = cli_utils.ReadOperatorsInfoFile(operatorsInfoPath)
+			if err != nil {
+				logger.Fatal(err.Error())
+			}
+		}
+		if err := viper.BindPFlag("operatorIDs", cmd.Flags().Lookup("operatorIDs")); err != nil {
+			logger.Fatal("😥 Failed to bind a flag: ", zap.Error(err))
+		}
+		oldParticipants := viper.GetStringSlice("operatorIDs")
+		if oldParticipants == nil {
+			logger.Fatal("😥 Failed to get operator IDs flag value: ", zap.Error(err))
```

Comment:
most of this code is repeated, for example loading the output path and verifying it is the same code as `initator/init.go`, should be take out to a separate function (probably in the `initiator` package) and used in both,
same goes for operators info/info path.
participants code is just a tad different because of the new and old participants here, but it already has loadParticipants for that.